_posts/2016-11-04-serverless-freeze-demonstration.md

```diff
@@ -0,0 +1,113 @@
+  arrayB.push("B");
+  arrayC.push("C");
+  arrayD.push("D");
+
+  console.log("Array A:", arrayA);
+  console.log("Array B:", arrayB);
+  console.log("Array C:", arrayC);
+  console.log("Array D:", arrayD);
+
+  arrayA = [];
+  arrayC = [];
+
+  callback(null);
+};
+```
+
+We initialize 2 empty arrays, arrayA and arrayB, outside of the handler.  When the function
+is run, these two arrays will remain in memory when the function is frozen.
+Inside of the hander we've got 2 more empty arrays, arrayC and arrayD.  During the
+execution of the function we simply push a string item to it and for testing, we
+clear our arrays arrayA and arrayC.  Let's look at what this does when AWS invokes this
+function three times:
+
+```bash
+serverless-freezing-demonstration % serverless invoke -f test -s dev -l
+null
+--------------------------------------------------------------------
+START RequestId: 778bcf95-a285-11e6-aa5a Version: $LATEST
+2016-11-04 07:54:40.248	778bcf95-a285-11e6-aa5a	Array A: [ 'A' ]
+2016-11-04 07:54:40.284	778bcf95-a285-11e6-aa5a	Array B: [ 'B' ]
+2016-11-04 07:54:40.284	778bcf95-a285-11e6-aa5a	Array C: [ 'C' ]
+2016-11-04 07:54:40.284	778bcf95-a285-11e6-aa5a	Array D: [ 'D' ]
+END RequestId: 778bcf95-a285-11e6-aa5a
+REPORT RequestId: 778bcf95-a285-11e6-aa5a	Duration: 37.54 ms	Billed Duration: 100 ms 	Memory Size: 128 MB	Max Memory Used: 14 MB
+
+serverless-freezing-demonstration % serverless invoke -f test -s dev -l
+null
+--------------------------------------------------------------------
+START RequestId: 7b760204-a285-11e6-bf4a Version: $LATEST
+2016-11-04 07:54:46.714 7b760204-a285-11e6-bf4a	Array A: [ 'A' ]
+2016-11-04 07:54:46.714 7b760204-a285-11e6-bf4a	Array B: [ 'B', 'B' ]
+2016-11-04 07:54:46.714 7b760204-a285-11e6-bf4a	Array C: [ 'C' ]
+2016-11-04 07:54:46.714 7b760204-a285-11e6-bf4a	Array D: [ 'D' ]
+END RequestId: 7b760204-a285-11e6-bf4a
+REPORT RequestId: 7b760204-a285-11e6-bf4a	Duration: 0.68 ms	Billed Duration: 100 ms 	Memory Size: 128 MB	Max Memory Used: 14 MB
+
+serverless-freezing-demonstration % serverless invoke -f test -s dev -l
+null
+--------------------------------------------------------------------
+START RequestId: 7ecbbbed-a285-11e6-99ff Version: $LATEST
+2016-11-04 07:54:52.288 7ecbbbed-a285-11e6-99ff	Array A: [ 'A' ]
+2016-11-04 07:54:52.289 7ecbbbed-a285-11e6-99ff	Array B: [ 'B', 'B', 'B' ]
+2016-11-04 07:54:52.289 7ecbbbed-a285-11e6-99ff	Array C: [ 'C' ]
+2016-11-04 07:54:52.289 7ecbbbed-a285-11e6-99ff	Array D: [ 'D' ]
+END RequestId: 7ecbbbed-a285-11e6-99ff
+REPORT RequestId: 7ecbbbed-a285-11e6-99ff	Duration: 0.52 ms	Billed Duration: 100 ms 	Memory Size: 128 MB	Max Memory Used: 14 MB
+```
+You'll notice in the first run, we have one pushed object to all four arrays.
+In the second run, arrayB has a second object in the array, and in the third
+run, arrayB continues to add the array.  arrayA doesn't exhibit this behavior
+because we clean him out prior to stopping the function.  arrayC and arrayD
```

Comment:
maybe instead of `clean him out`, say `because it is reinitialized to an empty array prior to stopping the function`.